### PR TITLE
Cleanup and simplify hashing modules.

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by shlevy** [comment]
Probably easiest to review commit-wise

File: hnix-store-core/src/System/Nix/ReadonlyStore.hs [hunk 1]
```diff
@@ -10,19 +10,20 @@ import qualified Data.HashSet as HS
 import           Data.Text (Text)
 import qualified Data.Text as T
 import           Data.Text.Encoding
-import           System.Nix.Internal.Hash
+import           System.Nix.Hash
 import           System.Nix.Path
 
 makeStorePath :: Text -> Text -> Digest 'SHA256 -> Text -> Path
```
Comment (mightybyte) [inline hnix-store-core/src/System/Nix/ReadonlyStore.hs]: Were you going to rename `Path` to `StorePath`?

Comment (shlevy) [reply hnix-store-core/src/System/Nix/ReadonlyStore.hs]: Yeah, I just scoped this original PR down to just the Hash module. After this is in I'll merge the ready open PRs then continue cleanup module by module

File: hnix-store-core/src/System/Nix/Internal/Hash.hs [hunk 2]
```diff
@@ -21,162 +17,129 @@ import qualified Crypto.Hash.SHA1       as SHA1
...
 
--- | A tag for different hashing algorithms
---   Also used as a type-level tag for hash digests
---   (e.g. @Digest SHA256@ is the type for a sha256 hash)
+-- | The universe of supported hash algorithms.
 --
---  When used at the type level, `n` is `Nat`
-data HashAlgorithm' n
+-- Currently only intended for use at the type level.
+data HashAlgorithm
```
Comment (mightybyte) [inline hnix-store-core/src/System/Nix/Internal/Hash.hs]: Nice simplification

**Pull request merged**